### PR TITLE
[operator] Manage Virtual Garden `kube-controller-manager` Deployment

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area dev-productivity delivery
/kind enhancement

**What this PR does / why we need it**:
With this PR, `gardener-operator` manages the virtual garden `kube-controller-manager` deployment.

Since the `serviceaccount-token` is always disabled, we don't need to support rotation of static `ServiceAccount` credentials.

The credentials rotation e2e test was adapt to support an advanced verifier for accessing the virtual garden cluster (similar to shoots).

**Which issue(s) this PR fixes**:
Part of #7016

**Release note**:

```feature operator
`gardener-operator` is now managing the `kube-controller-manager` instance as part of the virtual garden cluster control plane.
```

### CODE REVIEW

**Review by ScheererJ — COMMENTED** [review]
Thanks a lot for the change. It looks very good already.

I wonder though, why it was necessary to set `allowDangerousTypes=true` for `make generate`. If I understood it correctly the relevant line is about a float field which is already in for years. Did you understand what triggered this now?

**Comment by rfranzke** [comment]
> Thanks a lot for the change. It looks very good already.
> 
> I wonder though, why it was necessary to set `allowDangerousTypes=true` for `make generate`. If I understood it correctly the relevant line is about a float field which is already in for years. Did you understand what triggered this now?

The float was part of the `core.gardener.cloud` API only which is not part of a CRD but served by `gardener-apiserver`. With this PR, we embed the relevant API as part of the `operator.gardener.cloud` API which indeed is a CRD. Hence, it now appears for the first time in a CRD.

File: test/e2e/operator/garden/internal/rotation/virtual_garden_access.go [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/test/utils/access"
+)
+
+type clients struct {
+	accessSecret, clientCert, serviceAccountDynamic kubernetes.Interface
+}
+
+// VirtualGardenAccessVerifier uses the static token and admin kubeconfig to access the virtual garden.
```
Comment (shafeeqes) [inline test/e2e/operator/garden/internal/rotation/virtual_garden_access.go]: We are not using static token in the tests

**Review by shafeeqes — COMMENTED** [review]
Had a joint review with @ary1992.
Only one comment.

**Pull request merged**